### ISSUE TITLE
Scope/Reality Check re: Ember + Auth

### ISSUE THREAD

**Issue by kelvinma** [ga-wdi-boston/capstone-project#20]
Wouldn't mind getting a reality check on the scope of what I'm trying to do, re: Ember and Authentication. 

Currently trying to build an Ember app. Doesn't seem like it's going too badly without authentication, but once I try to dive in to authentication, the complexity seems to get exponentially greater, and not sure how to prioritize my concerns as they relate to the requirements of the project.

**Comment by jrhorn424** [comment]
Personally, I feel like you can focus on the main features of your app and implement auth later. Implement auth later is more complex than starting with auth, but usually because people forget to check which actions they want authorized.

I'd like @gaand and @tdyer to comment.

**Comment by cwcrane** [comment]
I have similar concerns to Kelvin's. Though I have a rails backend

**Comment by gaand** [comment]
I believe that if you use passport local, or token auth in rails, getting auth going won't add a lot of complexity.  I'm happy to chat about it.  Also,

**Comment by jrhorn424** [comment]
So, by default the Ember add ons that do auth on the front-end expect session auth, which you get if you use passport and express.

If you want to do token auth, you'll need to find an Ember add on that doesn't use sessions (see [my comment](https://github.com/ga-wdi-boston/portfolio-capstone-project/issues/13#issuecomment-164545764)), or you'll need to customize the adapter to add authorization headers to your requests.

I'm not sure which will be easier.

**Comment by kelvinma** [comment]
Yeah, the session auth seems like it's the way to go. I tried writing passport in to the November backend, and Antony helped me navigate its Express architecture. I got stuck trying to write passport to do the session store with Sequelize, since we did it in Mongo last time. I ran in to a lot of errors trying to get passport to talk to the database. 

On the front end, I read a lot of documentation on running simple auth for sessions while using service injections. I started up a dummy app, but it quickly became apparent to me that it was adding another layer of complexity since it was calling in controllers and didn't seem like it would play nicely with the pods structure. 

And then, of course, it's getting all of this stuff to play nicely together...

I think my strategy right now will be to just build my crappy Ember CRUD app, which won't really be all the useful without authentication (since I won't be able to have multiple users and secure any of their data), hopefully by this afternoon, and then concentrate on writing the server-side and client-side session store/auth, which is decidedly a stretch goal at this point. 

Does that seem like a reasonable strategy moving forward? (Don't even ask me about my git flow... Been pulled in so many directions, and so many false starts...)
